morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -463,13 +399,30 @@ public void ensureIndexes(final boolean background) {
         return getByKeys(null, keys);
     }
 
+    private DBCollection getCollection(final Object obj) {
+        if (obj == null) {
+            return null;
```

Comment:
That's only a valid argument if it's really ok for the user input itself to be null.  The one test that exercises this code path is TestDatastore#saveNull, but that's only because DatastoreImpl#save(T, com.mongodb.WriteConcern) doesn't check for a null entity until after calling DatastoreImpl#getCollection(java.lang.Object).